recipe-server/normandy/base/api/routers.py

```diff
@@ -0,0 +1,34 @@
+from django.conf.urls import url
+
+from rest_framework.routers import DefaultRouter
+
+from normandy.base.api.views import APIRootView
+
+
+class RouterWithDetachedViews(DefaultRouter):
+    """Router that allows for detached routes."""
```

Comment:
It's really unclear what "detached" means. I'd call this `MixedViewRouter` or something.

You might also consider splitting the API index stuff and the ability to register non-viewset views into two separate classes.